packages/material-ui/src/Popper/Popper.test.js

```diff
@@ -169,6 +167,32 @@ describe('<Popper />', () => {
     });
   });
 
+  describe('prop: keepMounted', () => {
+    it('should keep the children in the DOM', () => {
+      const wrapper = mount(<Popper {...defaultProps} open={false} keepMounted />);
+      assert.strictEqual(wrapper.find('span').length, 1);
+    });
+
+    /* Test case for https://github.com/mui-org/material-ui/issues/15180 */
+    it('should remove from DOM when closed whilst transition has entering status', () => {
+      const wrapper = mount(
+        <Popper {...defaultProps} open={false} keepMounted={false} transition>
+          {({ TransitionProps }) => (
+            <Grow {...TransitionProps}>
+              <span>Hello World</span>
+            </Grow>
+          )}
+        </Popper>,
+      );
+      wrapper.setProps({ open: true });
```

Comment:
Could you include an assertion before this or is this assertion actually for the mount+update behavior? In any case it would help knowing what the DOM looks like after the mount and before the update.

Reply:
This test is primarily for the mount+update behaviour. I didn't put an assertion before the `wrapper.setProps({ open: true })` as I believed this would (should) be covered by prior tests? I can most certainly put one in, though, to ensure that the children aren't mounted, if it would help to make it more understandable.

Reply:
Added quite a few additional assertions, to solidify the testing behaviour and ensure that the Transition component stays as `entering` and not to `entered` during the test.
I appreciate it's a quite complex text case, for an _odd_ issue, so am more than happy to add anything extra that's required to make this more understandable.